activerecord/CHANGELOG.md

```diff
@@ -1,3 +1,30 @@
+*   Add a way to index nested models' validation errors by their attribute.
+
+    Example:
+
+        class Violin < ActiveRecord::Base
+          has_many :tuning_pegs, index_errors: :string
+          accepts_nested_attributes_for :tuning_pegs
+        end
+
+        class TuningPeg < ActiveRecord::Base
+          belongs_to :violin
+          validates_numericality_of :pitch
+        end
+
+        violin = Violin.new(
+          tuning_pegs_attributes: [
+            { string: 1 },
+            { string: 2, pitch: 450.0 },
+            { string: 3, pitch: 440.0 },
+            { string: 4 }
```

Comment:
What should happen if `string` is not present in the attributes?

Reply:
@rafaelfranca thank you for review!
This is a good question, and I think that this way of using `index_errors` will only make sense under two conditions:
- indexing attribute is present
- indexing attribute is unique

Since we can't properly identify this record when indexing attribute value is nil, the error will be returned without an index (as if `index_errors` was set to false; in this case it would be `{"tuning_pegs.pitch":["is not a number"]}`). But if there is a `string` attribute presence validation, then it would be like: `{"tuning_pegs.pitch":["is not a number"], "tuning_pegs.string":["can't be blank"]}`, which will show the reason of missing index. 

This will also not be useful if indexing attribute is not unique for some reason, because the client itself will not be able to identify where some specific error belongs.

This is not perfect solution, and the whole PR's goal is easily achievable with 4-line method, so maybe it's not worth to add it, but I thought it would be a useful addition to index_errors.